test/e2e/apps/statefulset.go

```diff
@@ -1154,6 +1154,42 @@ var _ = SIGDescribe("StatefulSet", func() {
 		framework.ExpectNoError(err)
 		e2estatefulset.WaitForStatusAvailableReplicas(c, ss, 1)
 	})
+
+	ginkgo.It("MinReadySeconds should update availableReplicas accordingly when enabled [Feature:StatefulSetMinReadySeconds] [alpha]", func() {
+		ssName := "test-ss"
+		headlessSvcName := "test"
+		// Define StatefulSet Labels
+		ssPodLabels := map[string]string{
+			"name": "sample-pod",
+			"pod":  WebserverImageName,
+		}
+		ss := e2estatefulset.NewStatefulSet(ssName, ns, headlessSvcName, 2, nil, nil, ssPodLabels)
+		ss.Spec.MinReadySeconds = 10
+		setHTTPProbe(ss)
+		ss, err := c.AppsV1().StatefulSets(ns).Create(context.TODO(), ss, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		e2estatefulset.WaitForStatusAvailableReplicas(c, ss, 0)
```

Comment:
Why this 0? Do you want to show that the replicas won't be available immediately? If yes, this could be racy if reaching this statement would take more than 10 seconds.

Reply:
I did not want to put a longer time so the test can complete quickly.

There are two ways to solve this.
- make MinReadySeconds a bit higher - eg 30s? (test takes longer)
- remove the 0 check

Reply:
@ravisantoshgudimetla and yes I want to check the replicas are not available immediately

Reply:
Kube as a whole is eventually consistent. So, better to make the value [here](https://github.com/kubernetes/kubernetes/pull/104078/files#diff-43428d04b3bcd037c1b87eab7cfa08d529ecd179a80186f014d274b5ef5cc9f5R1167) longer like 30 seconds and wait till 5 or 10 seconds and make sure that the AvailableReplicas is not updated. Even this may be racy somtimes. I'd prefer having -ve tests in integration of units by faking clock.

Reply:
for now I put 30 there with the additional wait (thanks for the suggestion). I expect it not be that racy since we are just waiting for the controller to pick up the resource and set 0

Reply:
We can revisit the test if it is failing(flaking).

Reply:
I have similar concerns here as Ravi, this might be flaking when STS controller immediately creates the first replica and you'll never meet 0 status condition.

Reply:
This is not about the creation of a replica, but about the availability and since we set MinReadySeconds to 30 we should observe 0 for the first 30 seconds. WaitForStatusAvailableReplicas is waiting only 0-10s to observe the 0 to prevent the race